pkg/plugin/validator.go

```diff
@@ -19,19 +19,21 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
-	"net/url"
 	"regexp"
 	"strconv"
-	"strings"
 
 	"github.com/abcxyz/pkg/githubapp"
 	"github.com/google/go-github/v55/github"
 )
 
 const (
-	issueURLPatternRegExp = `^https:\/\/github.com\/([a-zA-Z0-9-]*)\/[a-zA-Z0-9-]*\/issues\/[0-9]+$`
+	regexpNamedGroupKeyOwner   = "owner"
+	regexpNameGroupKeyRepoName = "repoName"
+	regexpNamedGroupKeyIssue   = "issue"
 )
 
+var issueURLPatternRegExp = fmt.Sprintf(`^https:\/\/github.com\/(?P<%s>[a-zA-Z0-9-]*)\/(?P<%s>[a-zA-Z0-9-]*)\/issues\/(?P<%s>[0-9]+$)`, regexpNamedGroupKeyOwner, regexpNameGroupKeyRepoName, regexpNamedGroupKeyIssue)
```

Comment:
This will prevent us from supporting github on-prem. We should probably ignore everything before the domain part (https://github.com/) or parse the value as a URL (`url.Parse` as before) and do the validation on the `Path` attribute instead.

Reply:
For the on-prem case, can we still use the same github client to query the issue status?

Reply:
Yup - it's just a different base URL